browser/src/app/TilesMiddleware.ts

```diff
@@ -168,15 +197,17 @@ class Tile {
 	}
 
 	needsRehydration(): boolean {
-		return !this.imgDataCache && this.hasKeyframe();
+		if (this.rawDeltas.length === 0) return false;
+		const lastId = this.rawDeltas[this.rawDeltas.length - 1].id;
+		return this.lastPendingId !== lastId;
```

Comment:
I guess we don't reset the lastPendingId. Is it possible that needsRehydration returns true when a tile is made current? Even when the tile is already decompressed?